lib/webvalve.rb

```diff
@@ -33,16 +35,42 @@ def config_paths
       @config_paths ||= Set.new
     end
 
+    def logger
+      @logger ||= build_logger
+    end
+
+    def logger=(logger)
+      @logger = logger
+    end
+
+    def env
+      @env ||= (ENV['RAILS_ENV'] || ENV['RACK_ENV'] || 'development').inquiry
+    end
```

Comment:
because this is memoized and not reading from `Rails.env`, if someone were to stub `Rails.env` in a test, its value would not be reflected here.

we may want to consider checking `Rails.env if defined?(::Rails)` as well in here, and maybe not memoizing nor allowing override

Reply:
You could do something like this:

```ruby
if defined?(Rails)
  delegate :env, to: Rails
else
  def env
    @_env ||= (ENV['RAILS_ENV'] || ENV['RACK_ENV'] || 'development').inquiry
  end
end
```

Reply:
i.e. just don't memoize if you're delegating to Rails (which memoizes internally)